brouter-routing-app/src/main/AndroidManifest.xml

```diff
@@ -25,6 +25,51 @@
                 <action android:name="android.intent.action.MAIN" />
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
+            <!-- some apps (bluemail) do not recognize the .brf file extention, startactivity+intent is done for attachement with text/plain -->
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <data android:mimeType="text/plain"/>
+            </intent-filter>
+
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+                <category android:name="android.intent.category.LAUNCHER" />
+                <category android:name="android.intent.category.MULTIWINDOW_LAUNCHER" />
+            </intent-filter>
+            <!-- general intent to register .brf files -->
+            <intent-filter
+                android:label="@string/app_name"
+                android:priority="50">
+                <action android:name="android.intent.action.VIEW" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+                <data android:scheme="file"/>
+                <data android:scheme="content"/>
+                <data android:host="*"/>
+                <data android:pathPattern=".*\\.brf" />
+                <data android:pathPattern=".*\\..*\\.brf" />
+                <data android:pathPattern=".*\\..*\\..*\\.brf" />
+                <data android:pathPattern=".*\\..*\\..*\\..*\\.brf" />
+                <data android:pathPattern=".*\\..*\\..*\\..*\\..*\\.brf" />
+            </intent-filter>
+
+            <intent-filter
+                android:label="@string/app_name"
+                android:priority="50">
+                <action android:name="android.intent.action.VIEW" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+                <data android:scheme="file"/>
+                <data android:scheme="content"/>
+                <data android:host="*"/>
+                <data android:mimeType="*/*"/>
+                <data android:pathPattern=".*\\.brf" />
+                <data android:pathPattern=".*\\..*\\.brf" />
+                <data android:pathPattern=".*\\..*\\..*\\.brf" />
+                <data android:pathPattern=".*\\..*\\..*\\..*\\.brf" />
+                <data android:pathPattern=".*\\..*\\..*\\..*\\..*\\.brf" />
+            </intent-filter>
```

Comment:
This seems like a duplicate of the code block before - is it really necessary?

Reply:
> This seems like a duplicate of the code block before - is it really necessary?

Yes, all apps (as example osmand registration of .gpx files) do so by registration!
(it is not exactly a duplicate!!!)